src/adam_bashforth_adaptive.jl

```diff
@@ -0,0 +1,371 @@
+
+        successful_step = false
+        while !successful_step
+          #calculation (n+1)-th time position from n-th step size
+          t[n+1]=t[n]+h[n]
+          #####################################################################
+          #Calculating explicit approximatiion to y[n+1]
+          #####################################################################
+          #Current derivative: set dy[n] = F(t[n], y[n])
+          dy[n]=F(t[n], y[n])
+
+          #calculation of c_{j,q} coefficients for j=0,...,k-1; q = 0,...,k-1-(j-1)
+          g = g_coefs(g,c,h,t,n,min(n,steporder+3))
+
+          #calculation of b_j(n), Φstar_j(n), and Φ_j(n) coefficients from recurence relations
+          b, Φ, Φstar=Φ_and_Φstar_coefs(b, Φ, Φstar,dy[n],h,t,n,1:min(max(n-1,1),steporder+1))
+
+          ##Calculate next value, y[n+1], with order = k = steporder + 1. See
+          #equation (5.5) in Hairer et al, Vol 1, where we note that here
+          # y[n+1] = y_{k}(t_{n+1})
+          y[n+1] = y[n]
+          for j= 1:steporder
+            y[n+1] += h[n]*g[j]* Φstar[2,j]
+          end
+
+          dy[n+1] = F(t[n+1],y[n+1])
+          #NOTE:fixed step size explicit solver tested and gives correct result
+          ######################################################################
+          #CE Loop for implicit solver
+          ######################################################################
+          pred = y[n+1]
+          # Set CE_loops to be how many correction-evaulation cycles
+          # desired. As least one loop is necessary, for the sake of the
+          # adaptive step selection
+          CE_step = 1
+          i = 0
+          if i<CE_step && n>=steporder+1
+              i= i+1
+              #(C)orrection of y[n+1]
+              #add new coefficients  based on this prediction
+              ##computation of Φ, Φstar using recurrence relations
+              Φ = Φ_coefs(b, Φ, Φstar,dy[n+1],h,t,n+1,1:steporder+1)
+
+              ##calculation of g_{k}(n) coefficients using recurrence relations
+              ##has already been done above
+
+              # Harier equation 5.7
+              #pred = pred + h[n]*g[n,steporder+1]*Φ[n+1,steporder+1]
+              pred = pred + h[n]*g[steporder+1]*Φ[3,steporder+1]
+              #(E)valuate again
+              dy[n+1] = F(t[n+1],y[n+1])
+          end
+
+          #If `implicit` == true, we set the corrected prediction to be
+          #y[n+1]. If `implicit` == false, this is simply an explicit solver
+          implicit = true
+          if implicit
+              y[n+1] = pred
+          end
+          #NOTE:fixed step size implicit solver tested and gives correct result
+          ######################################################################
+          #Adaptive step selection
+          ######################################################################
+
+          #Don't use adaptive steps until n >= order + 2
+          if n<steporder+2
```

Comment:
I think it would be clearer:
```
if n<steporder+2
     h[n+1]=fac*h[n] # fac = .8 as default, by convention
    break
end
... 
```
Then there is no need for the `else` clause.